presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java

```diff
@@ -1209,6 +1219,66 @@ public String toString()
         }
     }
 
+    class BytesValuesExclusive
+            extends AbstractTupleDomainFilter
+    {
+        private final BytesValues delegate;
+
+        private BytesValuesExclusive(byte[][] values, boolean nullAllowed)
+        {
+            super(true, nullAllowed);
+            delegate = new BytesValues(values, nullAllowed);
+        }
+
+        public static BytesValuesExclusive of(byte[][] values, boolean nullAllowed)
+        {
+            return new BytesValuesExclusive(values, nullAllowed);
+        }
+
+        @Override
+        public boolean testBytes(byte[] value, int offset, int length)
+        {
+            return !delegate.testBytes(value, offset, length);
```

Comment:
check length first:

```
if (!delegate.testLength(length)) {
   return true;
}
```

Reply:
Sorry, got it lost during rebase, will add it back